huff_parser/src/lib.rs

```diff
@@ -185,6 +186,23 @@ impl Parser {
         std::mem::discriminant(&self.current_token.kind) == std::mem::discriminant(&kind)
     }
 
+    /// Checks if there is a duplicate macro name
+    pub fn check_duplicate_macro(
+        &self,
+        contract: &mut Contract,
+        m: MacroDefinition,
+    ) -> Result<(), ParserError> {
+        if contract.macros.binary_search_by(|_macro| _macro.name.cmp(&m.name)).is_ok() {
+            return Err(ParserError {
+                kind: ParserErrorKind::DuplicateMacro(m.name),
+                hint: Some("MACRO names should be unique".to_string()),
+                spans: AstSpan(vec![self.spans[2].clone()]),
```

Comment:
i assume this was meant to be m.spans rather than self.spans? 
Although i think the value being held will be the same

Reply:
rustc complains when I tried changing to m.spans..